tester/src/test/scala/spinal/tester/scalatest/SpinalSimStreamFifoTester.scala

```diff
@@ -31,31 +40,68 @@ class SpinalSimStreamFifoTester extends SpinalSimFunSuite {
 
       dut.io.flush #= false
 
+      var pushRatio, popRatio = 0.5
+      var sampledFLush = false
+      dut.clockDomain.onSamplings {
+        if(sampledFLush){
+          queueModel.clear()
+        }
+        if(Random.nextFloat() < 0.01){
+          pushRatio = 0.5 + 0.5*Random.nextFloat()
+          popRatio = 0.5 + 0.5*Random.nextFloat()
+        }
+        sampledFLush = dut.io.flush.toBoolean
+        assert(dut.io.occupancy.toInt == queueModel.size)
+        assert(dut.io.availability.toInt == depth - queueModel.size)
+        dut.io.flush #= Random.nextFloat() < 0.005
+      }
+
+
       //Push data randomly and fill the queueModel with pushed transactions
       dut.io.push.valid #= false
       dut.clockDomain.onSamplings {
         if (dut.io.push.valid.toBoolean && dut.io.push.ready.toBoolean) {
           queueModel.enqueue(dut.io.push.payload.toLong)
```

Comment:
since we update the testcase, should we use `ScoreboardInOrder` here?

Reply:
Baaaa, things are working already, i think it isn't necessary.